$OpenBSD: patch-src_expat_erl_c,v 1.1 2011/06/03 13:45:43 gsoares Exp $

Fix for SA44807, ejabberd Nested XML Entities Denial of Service Vulnerability.

from upstream git:
https://git.process-one.net/ejabberd/mainline/commit/bd1df027c622e1f96f9eeaac612a6a956c1ff0b6

--- src/expat_erl.c.orig	Mon Dec 13 08:21:26 2010
+++ src/expat_erl.c	Fri Jun  3 10:39:09 2011
@@ -113,6 +113,8 @@ static ErlDrvData expat_erl_start(ErlDrvPort port, cha
       d->parser, (XML_CharacterDataHandler)erlXML_CharacterDataHandler);
 
 
+   XML_SetDefaultHandler(d->parser, NULL);
+
    return (ErlDrvData)d;
 }
 
